### PR TITLE
Add new goodbye message syntax

### PR DESCRIPTION
**Pull request by smangio**
## Description
This PR will add the new goodbye message syntax.

If the question has a underscore block (`____`), replace the underscores with the answer.

`My name is _____ and im silly`

becomes

`Zelem says: My name is *LUKE* and im silly. (Ended by @smangio)`

Otherwise it'll just append the answer to the end of the question:

`Zelem say: Who is silly? *LUKE*. (Ended by @smangio)`

### CODE REVIEW

**Comment by spRite75** [comment]
> ## Description
> 
> This PR will add the new goodbye message syntax.
> 
> If the question has a underscore block (`____`), replace the underscores with the answer.
> 
> `My name is _____ and im silly`
> 
> becomes
> 
> `Zelem says: My name is *LUKE* and im silly. (Ended by @smangio)`
> 
> Otherwise it'll just append the answer to the end of the question:
> 
> `Zelem say: Who is silly? *LUKE*. (Ended by @smangio)`

really not happy about the discrimination in these docs

File: src/apps/zelem/index.ts [hunk 1]
```diff
@@ -116,10 +116,21 @@ export class Zelem {
...
+    const regexTest = /_{2,}/; // Finds an underscore block ('____');
+
+    const outputText = regexTest.test(this.currentQuestion)
+      ? this.currentQuestion.replace(regexTest, `*${this.currentMessage}*`)
+      : `${this.currentQuestion} *${this.currentMessage}*`;
+
     await this.wc.chat.postMessage({
       channel,
-      text: `Zelem says: *${this.currentMessage}*. (Ended by <@${byUser}>)`,
+      text: `Zelem says: ${outputText}${!promptHasPunctuation && '.'} (Ended by <@${byUser}>)`,
```
Comment (spRite75) [inline src/apps/zelem/index.ts]: regarding punctuation, a simpler way would be
```typescript
const outputText = regexTest.test(this.currentQuestion)
  ? this.currentQuestion.replace(regexTest, `*${this.currentMessage}*`)
  : `${this.currentQuestion} *${this.currentMessage}*.`;
```

Comment (spRite75) [reply src/apps/zelem/index.ts]: `promptHasPunctuation` is no longed needed, nor `questionLastChar`

File: src/apps/zelem/index.ts [hunk 2]
```diff
@@ -117,19 +117,16 @@ export class Zelem {
   private async sayGoodbye(byUser: string) {
     const channel = this.idsByChannel.get('weegee');
 
-    const questionLastChar = this.currentQuestion.substr(-1);
-    const promptHasPunctuation = questionLastChar === '.' || questionLastChar === '?';
-
     const regexTest = /_{2,}/; // Finds an underscore block ('____');
```
Comment (spRite75) [inline src/apps/zelem/index.ts]: could name this matchUnderscoreBlock?

Comment (smangio) [reply src/apps/zelem/index.ts]: sure

**Pull request merged**